sycl/test-e2e/Graph/invalid_event_wait.cpp

```diff
@@ -0,0 +1,29 @@
+// REQUIRES: level_zero, gpu
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// Tests that waiting on an event returned from a Record and Replay submission
+// throws.
+
+#include "graph_common.hpp"
+
+int main() {
+  queue Queue;
+
+  ext::oneapi::experimental::command_graph Graph{Queue.get_context(),
+                                                 Queue.get_device()};
+  Graph.begin_recording(Queue);
+
+  auto GraphEvent = Queue.submit(
+      [&](handler &CGH) { CGH.single_task<class TestKernel>([=]() {}); });
+
+  Graph.end_recording(Queue);
+
+  try {
+    GraphEvent.wait();
+  } catch (const sycl::exception &e) {
+    assert(e.code() == sycl::errc::invalid);
+  }
```

Comment:
You can have silent fail with this test. 
If no exception is thrown the test will pass because we do not enter into the catch branch, while it should fail.

Reply:
Fixed, thanks!